atomic_reactor/plugins/pre_resolve_remote_source.py

```diff
@@ -76,84 +78,70 @@ def run(self):
             return
 
         if (not get_allow_multiple_remote_sources(self.workflow)
-                and self.workflow.source.config.remote_sources):
+                and self.multiple_remote_sources_params):
             raise ValueError('Multiple remote sources are not supported, use single '
                              'remote source in container.yaml')
 
-        remote_source_params = self.workflow.source.config.remote_source
-        if not remote_source_params:
-            self.log.info('Aborting plugin execution: missing remote_source configuration')
+        if not (self.single_remote_source_params or self.multiple_remote_sources_params):
+            self.log.info('Aborting plugin execution: missing remote source configuration')
             return
 
         if self._dependency_replacements and not is_scratch_build(self.workflow):
             raise ValueError('Cachito dependency replacements are only allowed for scratch builds')
+        if self._dependency_replacements and self.multiple_remote_sources_params:
+            raise ValueError('Cachtio dependency replacements are not allowed for multiple remote sources')
+
+        remote_sources_workers_params = []
+        remote_sources_output = []
+        if self.multiple_remote_sources_params:
+            self.verify_multiple_remote_sources_names_are_unique()
+            for remote_source_params in self.multiple_remote_sources_params:
+                self.process_remote_source(
```

Comment:
this method waits for each cachito request, and processing em serially, and that will take forever

we want to first create all request and then wait for all of em to finish

Reply:
fixed